pkg/summarizer/summary_test.go

```diff
@@ -334,6 +339,91 @@ func TestUpdateDashboard(t *testing.T) {
 	}
 }
 
+func TestGroupPatherLookup(t *testing.T) {
+	mustPath := func(s string) *gcs.Path {
+		p, err := gcs.NewPath(s)
+		if err != nil {
+			t.Helper()
+			t.Fatalf("gcs.NewPath(%q) got error: %v", s, err)
+		}
+		return p
+	}
+
+	cases := []struct {
+		name      string
+		gp        groupPather
+		dash      string
+		tab       string
+		testgroup string
+		want      *gcs.Path
+		err       bool
+	}{
+		{
+			name: "empty",
+			want: &gcs.Path{},
+		},
+		{
+			name: "test group",
+			gp: groupPather{
+				configPath: *mustPath("gs://prefix/config"),
+				gridPrefix: "grid",
+			},
+			testgroup: "hello",
+			want:      mustPath("gs://prefix/grid/hello"),
+		},
+		{
+			name: "from tab state",
+			gp: groupPather{
+				configPath: *mustPath("gs://prefix/config"),
+				gridPrefix: "ignore",
+				tabPrefix:  "tabs",
+			},
+			dash:      "hi",
+			tab:       "there",
+			testgroup: "bad",
+			want:      mustPath("gs://prefix/tabs/hi/there"),
+		},
+		{
+			name: "interesting tab",
+			gp: groupPather{
+				configPath: *mustPath("gs://prefix/config"),
+				gridPrefix: "ignore",
+				tabPrefix:  "tabs",
+			},
+			dash:      "hi",
+			tab:       "My favorite tab!",
+			testgroup: "bad",
+			want: func() *gcs.Path {
```

Comment:
This function is a lot of complexity to make the internals of a url object compare right. I think it would be better to just compare path.String(); the behavior is what matters

Reply:
Added a TODO to implement a https://pkg.go.dev/github.com/google/go-cmp/cmp#Transform, which a good way to address this concern  (this is what protocmp.Transform() implements I suspect).

My main goal here is to fail something here when we change the tabulator function that this relies on. It is only one test so it doesn't seem necessary to change the rest of them when I can just adapt this one.

Reply:
I'm not sure I understand. Why would you want to avoid changing the other tests in `TestGroupPatherLookup`? Making "want" a `string` instead of `*gcs.Path` seems clear, unless you're worried about some side effect dependent on the internals of `gcs.Path`. And if you are, those tests should probably go with that package.